ckanext/s3filestore/uploader.py

```diff
@@ -151,12 +155,29 @@ def get_signed_url_to_key(self, key_path, expiredin=3600):
         # check whether the object exists in S3
         client.head_object(Bucket=self.bucket_name, Key=key_path)
 
+        current_time = time.time()
+        redis_conn = connect_to_redis()
+        cache_key = REDIS_PREFIX + key_path
```

Comment:
this cache key making should be a sub function as its used in two places that need to stay the same.

Reply:
we may need to encode the keypath to make it safe for redis

Reply:
Not necessary. From redis.io:

    Redis keys are binary safe, this means that you can use any binary sequence as a key, from a string like "foo" to the content of a JPEG file. The empty string is also a valid key.